src/bb/image-io/rt_u3v.h

```diff
@@ -311,9 +338,25 @@ class U3V {
 
         auto timeout_us = 30 * 1000 * 1000;
 
-        int32_t num_device = devices_.size();
+        int32_t num_device = num_sensor_;
         std::vector<ArvBuffer *> bufs(num_device);
 
+        if (sim_mode_){
```

Comment:
U3V class is getting messy. Simulation mode is better to be separated into different functions.